src/dmd/typesem.d

```diff
@@ -2783,56 +2775,49 @@ void resolve(Type mt, const ref Loc loc, Scope* sc, Expression* pe, Type* pt, Ds
                 e = e.expressionSemantic(sc);
                 mt.resolveExp(e, pt, pe, ps);
             }
+            if (*pt)
+                (*pt) = (*pt).addMod(mt.mod);
         }
-        if (*pt)
-            (*pt) = (*pt).addMod(mt.mod);
```

Comment:
Why was this moved?

Reply:
It was done differently on line 2766 of the PR.